src/R/Components/Impl/InteractiveWorkflow/Commands/InterruptRCommand.cs

```diff
@@ -11,47 +11,33 @@ public sealed class InterruptRCommand : IAsyncCommand {
         private readonly IRInteractiveWorkflow _interactiveWorkflow;
         private readonly IRSession _session;
         private readonly IDebuggerModeTracker _debuggerModeTracker;
-        private volatile bool _enabled;
 
         public InterruptRCommand(IRInteractiveWorkflow interactiveWorkflow, IDebuggerModeTracker debuggerModeTracker) {
             _interactiveWorkflow = interactiveWorkflow;
             _session = interactiveWorkflow.RSession;
             _debuggerModeTracker = debuggerModeTracker;
-
-            _session.Connected += OnConnected;
-            _session.Disconnected += OnDisconnected;
-
-            _session.BeforeRequest += OnBeforeRequest;
-            _session.AfterRequest += OnAfterRequest;
         }
 
-        private void OnConnected(object sender, RConnectedEventArgs e) => _enabled = true;
-        private void OnDisconnected(object sender, EventArgs e) => _enabled = false;
-
-        private void OnBeforeRequest(object sender, RBeforeRequestEventArgs e) {
-            _enabled = e.Contexts.Count != 1; // Disable command only if prompt is in the top level
-        }
-
-        private void OnAfterRequest(object sender, RAfterRequestEventArgs e) => _enabled = true;
-
         public CommandStatus Status {
             get {
                 var status = CommandStatus.Supported;
                 if (_interactiveWorkflow.ActiveWindow == null) {
                     status |= CommandStatus.Invisible;
-                } else if (_session.IsHostRunning && _enabled && !_debuggerModeTracker.IsInBreakMode) {
+                } else if (CanInterrupt()) {
                     status |= CommandStatus.Enabled;
                 }
                 return status;
             }
         }
 
         public async Task InvokeAsync() {
-            if (_enabled) {
-                _enabled = false;
+            if (CanInterrupt()) {
                 _interactiveWorkflow.Operations.ClearPendingInputs();
                 await _session.CancelAllAsync();
             }
         }
+
+        private bool CanInterrupt()
+            => _session.IsHostRunning && _session.IsProcessing && !_session.IsReadingUserInput && !_debuggerModeTracker.IsInBreakMode;
```

Comment:
Wait, I missed the other comment. So the real problem isn't that the host can't handle the interrupt, it's that the Interactive Window doesn't support it when reading input? Can we send it dummy input (just plain Enter) to get past the prompt it's stuck on?